apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/bootstrap/BootstrapInstrumentBoost.java

```diff
@@ -160,29 +173,49 @@ private static boolean prepareJREInstrumentation(PluginFinder pluginFinder,
      * One key step to avoid class confliction between AppClassLoader and BootstrapClassLoader
      *
      * @param classesTypeMap hosts injected binary of generated class
-     * @param classPool to generate new class
+     * @param typePool to generate new class
      * @param templateClassName represents the class as template in this generation process. The templates are
      * pre-defined in SkyWalking agent core.
      * @param methodsInterceptor
      */
-    private static void generateDelegator(Map<String, byte[]> classesTypeMap, ClassPool classPool,
+    private static void generateDelegator(Map<String, byte[]> classesTypeMap, TypePool typePool,
         String templateClassName, String methodsInterceptor) {
         String internalInterceptorName = internalDelegate(methodsInterceptor);
         try {
-            CtClass interClass = classPool.get(templateClassName);
-            interClass.setName(internalInterceptorName);
-
-            CtField interceptorDefine = interClass.getField("TARGET_INTERCEPTOR");
-            interClass.removeField(interceptorDefine);
-
-            interceptorDefine = CtField.make("private static String TARGET_INTERCEPTOR = \"" + methodsInterceptor + "\";", interClass);
-            interClass.addField(interceptorDefine);
+            TypeDescription templateTypeDescription = typePool.describe(templateClassName)
+                .resolve();
+
+            DynamicType.Unloaded interceptorType = new ByteBuddy()
+                .redefine(templateTypeDescription, ClassFileLocator.ForClassLoader.of(BootstrapInstrumentBoost.class.getClassLoader()))
+                .name(internalInterceptorName)
+                .field(named("TARGET_INTERCEPTOR")).value(methodsInterceptor)
+                .make();
+
+            classesTypeMap.put(internalInterceptorName, interceptorType.getBytes());
+
+            try {
+                File debuggingClassesRootPath = null;
```

Comment:
wirte debug class when Config.Agent.IS_OPEN_DEBUGGING_CLASS is true?

Reply:
Yes. It exists before, I just move it to support this new part.

Reply:
Yeah,I mean I think it is not necessary to write the class to disk every time and It seems duplicated in this place.
![image](https://user-images.githubusercontent.com/11530760/62008921-21f84500-b18b-11e9-854c-ed404d163d5a.png)

Reply:
I will remove the duplicated ones.

Reply:
I removed the duplicated file write and map#put. Please recheck, if it is OK now, please approve again.